The portal widget and mpsc::Receiver<>

I have still absolutely no idea for the reason for the issue. Of course it is not really important for the chess GUI, but it would be nice to understand the problem, and to know if there might exist an idea how it can be solved later. Because message passing with background tasks is something that a few serious Xilem apps might want to do.

https://github.com/StefanSalewski/xilem-chess

```
stefan@hx90 /tmp/xilem-chess/src $ diff ~/allgitprojects/xilem-chess/src/main.rs main.rs 
21c21
<         button, checkbox, flex_col, flex_row, grid, label, prose, sized_box, slider, task,
---
>         button, checkbox, flex_col, flex_row, grid, label, portal, prose, sized_box, slider, task,
515c515
<         main_layout(state),
---
>         portal(main_layout(state)),
stefan@hx90 /tmp/xilem-chess/src $ cargo build
   Compiling xilem-chess v0.5.0 (/tmp/xilem-chess)
error[E0277]: `std::sync::mpsc::Receiver<engine::Move>` cannot be shared between threads safely
   --> src/main.rs:513:39
    |
513 |   fn app_logic(state: &mut AppState) -> impl WidgetView<Edit<AppState>> + use<> {
    |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `std::sync::mpsc::Receiver<engine::Move>` cannot be shared between threads safely
514 | /     fork(
515 | |         portal(main_layout(state)),
516 | |         state.active.then(|| {
517 | |             task(
...   |
529 | |         }),
530 | |     )
    | |_____- return type was inferred to be `Fork<_, _, Portal<impl WidgetView<&mut AppState>, &mut AppState, ()>, Option<Task<_, (), {closure@...}, ..., ()>>>` here
    |
    = help: within `Fork<&mut AppState, (), Portal<impl WidgetView<&mut AppState>, &mut AppState, ()>, Option<...>>`, the trait `Sync` is not implemented for `std::sync::mpsc::Receiver<engine::Move>`
note: required because it appears within the type `Option<std::sync::mpsc::Receiver<engine::Move>>`
   --> /usr/lib/rust/1.90.0/lib/rustlib/src/rust/library/core/src/option.rs:593:10
    |
593 | pub enum Option<T> {
    |          ^^^^^^
note: required because it appears within the type `AppState`
   --> src/main.rs:138:8
    |
138 | struct AppState {
    |        ^^^^^^^^
    = note: required because it appears within the type `&mut AppState`
    = note: required because it appears within the type `(&mut AppState, ())`
note: required because it appears within the type `PhantomData<(&mut AppState, ())>`
   --> /usr/lib/rust/1.90.0/lib/rustlib/src/rust/library/core/src/marker.rs:823:12
    |
823 | pub struct PhantomData<T: PointeeSized>;
    |            ^^^^^^^^^^^
note: required because it appears within the type `xilem::view::Portal<impl WidgetView<&'static mut AppState>, &mut AppState, ()>`
   --> /home/stefan/.cargo/git/checkouts/xilem-420a6f61d3d10f60/c2ab850/xilem/src/view/portal.rs:27:12
    |
 27 | pub struct Portal<V, State, Action> {
    |            ^^^^^^
note: required because it appears within the type `Fork<&mut AppState, (), Portal<impl WidgetView<&mut AppState>, &mut AppState, ()>, Option<...>>`
   --> /home/stefan/.cargo/git/checkouts/xilem-420a6f61d3d10f60/c2ab850/xilem_core/src/views/fork.rs:29:12
    |
 29 | pub struct Fork<State, Action, Active, Alongside> {
    |            ^^^^
    = note: required for `Fork<&mut AppState, (), Portal<impl WidgetView<&mut AppState>, &mut AppState, ()>, Option<...>>` to implement `WidgetView<&'static mut AppState>`
    = note: the full name for the type has been written to '/tmp/xilem-chess/target/debug/deps/xilem_chess-af74ad5cf39d9066.long-type-2128047968465995106.txt'
    = note: consider using `--verbose` to print the full type name to the console

For more information about this error, try `rustc --explain E0277`.
error: could not compile `xilem-chess` (bin "xilem-chess") due to 1 previous error
stefan@hx90 /tmp/xilem-chess/src $ 
```

--- a/xilem_masonry/src/view/portal.rs
+++ b/xilem_masonry/src/view/portal.rs
@@ -26,7 +26,7 @@ where
 #[must_use = "View values do nothing unless provided to Xilem."]
 pub struct Portal<V, State, Action> {
     child: V,
-    phantom: PhantomData<(State, Action)>,
+    phantom: PhantomData<fn(State) -> Action>,
 }
 
 impl<V, State, Action> ViewMarker for Portal<V, State, Action> {}

